src/storage/src/hummock/compactor/mod.rs

```diff
@@ -765,51 +726,10 @@ pub fn start_shared_compactor(
-                                                vacuum_task: Some(vacuum_task),
-                                            };
-                                            match cloned_grpc_proxy_client.report_vacuum_task(report_vacuum_task_request).await {
-                                                Ok(_) => tracing::info!("Finished vacuuming SSTs"),
-                                                Err(e) => tracing::warn!(error = %e.as_report(), "Failed to report vacuum task"),
-                                            }
-                                        }
-                                        Err(e) => {
-                                            tracing::warn!(error = %e.as_report(), "Failed to vacuum task")
-                                        }
-                                    }
+                                    tracing::error!(?vacuum_task, "unexpected vacuum task");
                                 }
                                 dispatch_compaction_task_request::Task::FullScanTask(full_scan_task) => {
-                                    let start_after = full_scan_task.start_after.clone();
-                                    match Vacuum::handle_full_scan_task(full_scan_task, context.sstable_store.clone())
-                                        .await
-                                    {
-                                        Ok((object_ids, total_object_count, total_object_size, next_start_after)) => {
-                                            let report_full_scan_task_request = ReportFullScanTaskRequest {
-                                                object_ids,
-                                                total_object_count,
-                                                total_object_size,
-                                                next_start_after,
-                                                start_after,
-                                            };
-                                            match cloned_grpc_proxy_client
-                                                .report_full_scan_task(report_full_scan_task_request)
-                                                .await
-                                            {
-                                                Ok(_) => tracing::info!("Finished full scan SSTs"),
-                                                Err(e) => tracing::warn!(error = %e.as_report(), "Failed to report full scan task"),
-                                            }
-                                        }
-                                        Err(e) => {
-                                            tracing::warn!(error = %e.as_report(), "Failed to iter object");
-                                        }
-                                    }
+                                    tracing::error!(?full_scan_task, "unexpected scan task");
```

Comment:
I think it's acceptable to use unreachable! directly, it's easier to find errors. 

Compatibility issues would also only occur when switching cluster versions.